### ISSUE TITLE
Team Project ('primary repository')

### ISSUE THREAD

**Issue by bobheadxi** [ubclaunchpad/rocket2#177]
The new UBC Launch Pad website will depend on a few presumed features that would be very nice to have:

* each `team` *must* have a single `project`, which takes the form of a GitHub repository. This repository might be private or public.
* Rocket should be able to return the following things about a `project`:
     * `display_name`: this should default to the GitHub repo name. editable
     * `short_description`: this should be the GitHub repo description. not editable.
     * `long_description`: blank by default.  editable. capped at maybe... 300 characters?
     * `tags`: these should be the GitHub repo tags. not editable.
     * `languages`: should be available via GitHub API. not editable.
     * `github_link`: required. link to repository. not editable.
     * `website_link`: blank by default. link to website. editable.
     * `medium_link`: blank by default. link to medium. editable.
     * `appstore_link`: blank by default. link to apple app store. editable.
     * `playstore_link`: blank by default. link to google play store. editable.

Some problems with this:
* doesn't play well with multi-repo, ie @bfbachmann 's project. If we want to support these properly, I would make `github_link` an array and join the `tags` of all repos, but no other changes.

This is probably blocked by a bunch of other stuff that Rocket needs to implement first, and the website design isn't even finalized yet, so this is *relatively* low-priority, though something to keep in mind. I'll be happy to help implement this later on whenever yall think Rocket is ready for it 😄 Thanks!

**Comment by CoffeeTableEspresso** [comment]
We could do `github_link` as PK.

What's the point of `long_description`?

Does changing `display_name` update the GitHub repo name?

Will there ever be a case where 2 teams own a project? If not, we can easily query to Projects table based on the team that owns it.

**Comment by omar2535** [comment]
Sub tickets to make:

- Make relevant models / tables
- Slack commands & github API
- Webhooks

**Comment by rwblickhan** [comment]
Note: when we delete a team, we will need to delete references to the team in projects.

**Comment by bobheadxi** [comment]
I've added a `languages` field to the spec - thanks guys!

**Comment by cheukyin699** [comment]
![20190202_120201](https://user-images.githubusercontent.com/6824907/52527197-8aa14300-2c79-11e9-8ac2-1000ac7f5c94.jpg)

Let me just put this here~

**Comment by cheukyin699** [comment]
I'm pretty sure this is implemented. Pretty experimental though. Closing.
